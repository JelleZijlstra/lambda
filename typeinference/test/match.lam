type foo = Foo int | Bar bool in
let f = \x. match x with {
	Foo n -> print n
	| Bar b -> print (if b then 1 else 0) } in
(f (Foo 3), f (Bar false))