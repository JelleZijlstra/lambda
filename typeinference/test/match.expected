%%stdout%%
3
0
Result: module
	let __result__ = ((), ())
end
%%stderr%%
